etsin_finder/frontend/__tests__/components/general/qvain.translationTabInputModal.test.jsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react'
+import { shallow } from 'enzyme'
+import 'chai/register-expect'
+import { configure } from 'mobx'
+
+import TranslationTabInputModal, {
+  TranslationTabInputElem,
+} from '../../../js/components/qvain/general/input/translationTabInputModal'
+import FieldClass from '../../../js/stores/view/qvain/qvain.field'
+import { buildStores } from '../../../js/stores'
+
+configure({ safeDescriptors: false })
+
+const template = () => ({})
+const model = () => ({})
+
+describe('TranslationsTabInputModal', () => {
+  let wrapper, inputElement
+  let Field, Stores
+
+  let props
+
+  describe('given all props', () => {
+    const render = extraProps => {
+      Stores = buildStores()
+      Field = new FieldClass(Stores.Qvain, template, model)
+      Field.create()
+
+      props = {
+        Field,
+        datum: 'datum',
+        translationsRoot: 'translationsRoot',
+        language: 'en',
+        handleBlur: jest.fn(),
+      }
+
+      const parsedProps = { ...props, ...extraProps }
+
+      wrapper = shallow(<TranslationTabInputModal {...parsedProps} />)
+      inputElement = wrapper.find('#datumField')
```

Comment:
either that or we could use some other prop that is locally unique (locally meaning inside a component that is currently being tested). In this case it could be:

```
inputElement = wrapper.findWhere(elem => elem.prop("component") === TranslationTabInputElem)
```

Overalls Translation components usually have some characteristics that differs from other Translation components.

using props is sometimes little bit confusing because component not necessary tell anything about the component. That's way harness has label. It gives understandable name to the component. Ofc it also gives a nice opportunity to use restoreWrapper functionality.

Reply:
refactored test to use findWhere instead of id.